File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/infrastruktur/modellierung/de/AggregatkennungSerialisierer.java
package org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;

import java.io.IOException;

public class AggregatkennungSerialisierer extends StdSerializer<Aggregatkennung>
{
    public AggregatkennungSerialisierer() {
        this(null);
    }

    public AggregatkennungSerialisierer(final Class<Aggregatkennung> t) {
        super(t);
    }

    @Override
    public void serialize(
            final Aggregatkennung value,
            final JsonGenerator jgen,
            final SerializerProvider provider)
            throws IOException
    {
        jgen.writeString(value.wert().toString());
    }
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/infrastruktur/modellierung/de/AggregatkennungSerialisierer.kt
package org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.ser.std.StdSerializer

import java.io.IOException

class AggregatkennungSerialisierer @JvmOverloads constructor(t: Class<Aggregatkennung>? = null) : StdSerializer<Aggregatkennung>(t) {

    @Throws(IOException::class)
    override fun serialize(
            value: Aggregatkennung,
            jgen: JsonGenerator,
            provider: SerializerProvider) {
        jgen.writeString(value.id.toString())
    }
}
